src/audio_sink.rs

```diff
@@ -53,5 +55,78 @@ mod portaudio_sink {
     }
 }
 
+#[cfg(feature = "portaudio-sink")]
 pub type DefaultSink = portaudio_sink::PortAudioSink<'static>;
 
+#[cfg(feature = "pulseaudio-sink")]
+mod pulseaudio_sink {
+    use audio_sink::Sink;
+    use std::io;
+    use libpulse_sys::*;
+    use std::ptr::{null, null_mut};
+    use std::mem::{transmute};
+
+    pub struct PulseAudioSink(*mut pa_simple);
+
+    #[cfg(target_arch = "arm")]
+    type PtrType = u8;
+
+    #[cfg(target_arch = "x86_64")]
+    type PtrType = i8;
+
+    #[cfg(target_arch = "x86")]
+    type PtrType = i8;
+    
+    
+    impl PulseAudioSink {
+        pub fn open() -> PulseAudioSink {
+            println!("Using PulseAudioSink");
+
+            let ss = pa_sample_spec {
+                format: PA_SAMPLE_S16LE,
+                channels: 2, // stereo
+                rate: 44100
+            };
+            
+            let s = unsafe {
+                pa_simple_new(null(),             // Use the default server.
+                              "librespot".as_ptr() as *const PtrType,  // Our application's name.
```

Comment:
This isn't safe, see https://doc.rust-lang.org/std/ffi/struct.CString.html for how to pass strings to C functions